tests/src/Interop/PInvoke/SafeHandles/SafeHandleNative.cs

```diff
@@ -89,6 +89,9 @@ public object MarshalNativeToManaged(IntPtr pNativeData)
         [DllImport(nameof(SafeHandleNative))]
         public static extern void GetHandleAndCookie(IntPtr value, out TestSafeHandle handle, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(ThrowingCustomMarshaler), MarshalCookie = "")] out object cookie);
 
+        [DllImport(nameof(SafeHandleNative))]
+        public static extern void GetHandleAndArray(out short arrSize, [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 0)] out short[] arrShort, IntPtr value, out TestSafeHandle handle);
```

Comment:
Note that `GetHandleAndArray` has different order of parameters from `GetHandleAndCookie` above. I'm not sure that one is actually testing reliable unmarshaling - the SafeHandle argument would already be unmarshalled by the time we unmarshal the throwing cookie.

I can fix that if you agree it's a test bug.

Reply:
Yeah I think it might be a test bug. If you could fix that, I'd appreciate it.